### PR TITLE
Let GreeterEnvironment option take the precedence over PAM environment

### LINKED ISSUE #1758: Should the PAM's default environment override the GreeterEnvironment option?
pam_putEnv doesn't override PAM's environment loaded with pam_env.so so [this](https://github.com/sddm/sddm/blob/07631f2ef00a52d883d0fd47ff7d1e1a6bc6358f/src/daemon/Greeter.cpp#L183-L191) and [this](https://github.com/sddm/sddm/blob/07631f2ef00a52d883d0fd47ff7d1e1a6bc6358f/src/helper/HelperApp.cpp#L170-L181) gets overriden thanks to [this](https://github.com/sddm/sddm/blob/07631f2ef00a52d883d0fd47ff7d1e1a6bc6358f/src/helper/backend/PamBackend.cpp#L276). Is all that intended or should the imported environment variables from sddm process and/or GreeterEnvironment options take the precedence?

### PR DESCRIPTION
**Pull request by ilya-fedin**
Fixes #1758

### CODE REVIEW

**Comment by Vogtinator** [comment]
On one hand having some greeter specific code in backend.cpp is wrong, but on the other hand it's at least set in the same place as other environment variables.

The environment handling is spaghetti anyways so it doesn't really make a difference.

I'll have a closer look next week, but from a quick glance it looks good.

**Comment by ilya-fedin** [comment]
> I'll have a closer look next week

ping? :thinking:

File: src/helper/HelperApp.cpp [hunk 1]
```diff
@@ -167,19 +167,6 @@ namespace SDDM {
...
-                    qWarning() << "Malformed environment variable" << entry;
-                    continue;
-                }
-                env.insert(entry.left(index), entry.mid(index + 1));
-            }
-        }
-
         if (!m_session->path().isEmpty()) {
             env.insert(m_session->processEnvironment());
             m_session->setProcessEnvironment(env);
```
Comment (Vogtinator) [inline src/helper/HelperApp.cpp:172]: What do you think about just moving those assignments here, after the `env.insert(m_session->processEnvironment());` call? That should have the same effect but is IMO a slightly more fitting place

Comment (ilya-fedin) [reply src/helper/HelperApp.cpp:172]: Not entirely the same, `getpwnam` and `login_getpwclass` will have the priority over GreeterEnvironment but will still help with the exact case of QT_PLUGIN_PATH in PAM though.

Comment (Vogtinator) [reply src/helper/HelperApp.cpp:172]: Yeah, I don't really know what `setclassenvironment` does on FreeBSD. Maybe it should override GreeterEnvironment, maybe it shouldn't...

Comment (ilya-fedin) [reply src/helper/HelperApp.cpp:172]: I'd say it has sense for GreeterEnvironment to override anything else

Comment (Vogtinator) [reply src/helper/HelperApp.cpp:172]: Ok.

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**